mlos_bench/setup.py

```diff
@@ -24,33 +25,44 @@
     warning(f"setuptools_scm failed to find git version, using version from _version.py: {e}")
 
 
-extra_requires = {
+extra_requires: Dict[str, List[str]] = {
     # Additional tools for extra functionality.
-    'azure': 'azure-storage-file-share',
-    'sqlalchemy': 'sqlalchemy',
+    'azure': ['azure-storage-file-share'],
+    'storage-sql-duckdb': ['sqlalchemy', 'duckdb_engine'],
+    'storage-sql-mysql': ['sqlalchemy', 'mysql-connector-python'],
+    'storage-sql-postgres': ['sqlalchemy', 'psycopg2'],
+    'storage-sql-sqlite': ['sqlalchemy'],   # sqlite3 comes with python, so we don't need to install it.
     # Transitive extra_requires from mlos-core.
-    'emukit': 'emukit',
-    'skopt': 'scikit-optimize',
+    'emukit': ['emukit'],
+    'skopt': ['scikit-optimize'],
 }
 
 # construct special 'full' extra that adds requirements for all built-in
 # backend integrations and additional extra features.
-extra_requires['full'] = list(set(chain(extra_requires.values())))  # type: ignore[assignment]
+extra_requires['full'] = list(set(chain(*extra_requires.values())))
+
+extra_requires['full-tests'] = extra_requires['full'] + [
+    'pytest',
+    'pytest-forked',
+    'pytest-xdist',
+    'pytest-cov',
+    'pytest-local-badge',
```

Comment:
Moves pytest related packages to setup.py instead of conda so that the package can be tested independently of the conda env. Also reduces some of the places we have duplicate packages listed.